### DISCUSSION TITLE
fetch method not update that data property

### DISCUSSION THREAD

**Discussion by msm082919** [nuxt/nuxt#10258] [Questions (Nuxt 2)] [answered]
```html
<template>
  <div>
    <h1>Index</h1>
    <button @click="showMe">BUTTON</button>
  </div>
</template>

<script>
export default {
  name: 'Index',

  async fetch ({ $axios }) {
    const result = await $axios.$get('https://api.sampleapis.com/beers/ale')
    console.log(result) // show arr that length 100.
    this.arr = result
  },

  data: () => ({
    arr: []
  }),

  methods: {
    showMe() {
      console.log(this.arr) // show arr that length 0.
    }
  }
}
</script>
```

The [example on the official website](https://nuxtjs.org/docs/features/data-fetching) is the same way.

But my this.arr is not updated.

What should I do to solve this problem?

This really drives me crazy. 🤯🤯🤯🤯🤯🤯🤯🤯🤯🤯🤯🤯🤯🤯

why? Why the hell?

My Nuxt version is 2.15.8

**Comment by danielroe** [comment, marked as answer]
See the first note here: https://nuxtjs.org/docs/features/data-fetching/#the-fetch-hook

tl;dr: Rather than accessing axios from the context, use `this.$axios`.

**Reply by msm082919** [reply]
thank you 😊
